docs/reference/scripting/access-fields.asciidoc

```diff
@@ -0,0 +1,87 @@
+[[script-fields-api]]
+== Access fields in a document with the `field` API
+++++
+<titleabbrev>Access fields in a document</titleabbrev>
+++++
+
+beta::["The `field` API is still in development and should be considered a beta feature. The API is subject to change and this iteration is likely not the final state.",{es-issue}78920]
+
+Use the `field` API to access document fields:
+
+[source,painless]
+----
+field('my_field').get(<default_value>)
+----
+
+This API fundamentally changes how you access documents in Painless. Previously,
+you had to access the `doc` map with the field name that you wanted to access:
+
+[source,painless]
+----
+doc['my_field'].value
+----
+
+Accessing document fields this way didn't handle missing values or missing
+mappings, which meant that to write robust Painless scripts, you needed to
+include logic to check that both fields and values exist.
+
+Instead, use the `field` API, which is the preferred approach to access
+documents in Painless. The `field` API handles missing values, and will evolve
+to abstract access to `_source` and `doc_values`.
+
+NOTE: Some fields aren't yet compatible with the `fields` API, such as `text` or
+`geo` fields. Continue using `doc` to access field types that the `field` API
+doesn't support.
+
+The `field` API returns a `Field` object that iterates over fields with 
+multiple values, providing access to the underlying value through the
+`get(<default_value>)` method, as well as type conversion and helper methods. 
+
+The `field` API returns the default value that you specify, regardless of
+whether the field exists or has any values for the current document.
+This means that the `field` API can handle missing values without requiring 
+additional logic. For a reference (`Object`) type such as `keyword`, the default 
```

Comment:
Lets remove (`Object`) here because there is an Object type and that could be confusing.